src/NServiceBus.Core/Installation/InstallationComponent.cs

```diff
@@ -10,72 +10,117 @@
 
-            shouldRunInstallers = settings.GetOrDefault<bool>("Installers.Enable");
-
-            if (!shouldRunInstallers)
+            if (!configuration.ShouldRunInstallers)
             {
-                return;
+                return component;
             }
 
             foreach (var installerType in concreteTypes.Where(t => IsINeedToInstallSomething(t)))
             {
                 containerComponent.ContainerConfiguration.ConfigureComponent(installerType, DependencyLifecycle.InstancePerCall);
             }
+
+            return component;
         }
 
         public async Task Start()
         {
-            if (!shouldRunInstallers)
+            if (!configuration.ShouldRunInstallers)
             {
                 return;
             }
 
-            var queueBindings = settings.Get<QueueBindings>();
-            var username = GetInstallationUserName();
+            var installationUserName = GetInstallationUserName();
 
-            if (settings.CreateQueues())
+            if (configuration.ShouldCreateQueues)
             {
-                await receiveComponent.CreateQueuesIfNecessary(queueBindings, username).ConfigureAwait(false);
+                await receiveComponent.CreateQueuesIfNecessary(queueBindings, installationUserName).ConfigureAwait(false);
             }
 
             foreach (var installer in containerComponent.Builder.BuildAll<INeedToInstallSomething>())
             {
-                await installer.Install(username).ConfigureAwait(false);
+                await installer.Install(installationUserName).ConfigureAwait(false);
             }
         }
 
         string GetInstallationUserName()
         {
-            if (!settings.TryGet("Installers.UserName", out string userName))
+            if (configuration.InstallationUserName != null)
             {
-                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
-                {
-                    userName = $"{Environment.UserDomainName}\\{Environment.UserName}";
-                }
-                else
-                {
-                    userName = Environment.UserName;
-                }
+                return configuration.InstallationUserName;
+            }
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                return $"{Environment.UserDomainName}\\{Environment.UserName}";
             }
 
-            return userName;
+            return Environment.UserName;
         }
 
-        ReadOnlySettings settings;
+        Configuration configuration;
         ContainerComponent containerComponent;
         ReceiveComponent receiveComponent;
-        bool shouldRunInstallers;
+        QueueBindings queueBindings;
 
         static bool IsINeedToInstallSomething(Type t) => typeof(INeedToInstallSomething).IsAssignableFrom(t);
+
+        public class Configuration
+        {
+            public Configuration(SettingsHolder settings)
```

Comment:
As long as it will throw a meaningful exception, no need to go that far.

Reply:
Would there be a benefit to use the ReadonlySettings thingy?

Reply:
We would have to split into 2 separate classes since we need to write to the settings while we are "configuring". Given that the settings will throw if we try to write them after they are "locked" I'm thinking this is YAGNI for now until we see how this plays out?

@timbussmann we talked about this, are you still happy with that approach?